src/http_server/api/v1/metrics.c

```diff
@@ -397,15 +397,45 @@ void cb_metrics_prometheus(mk_request_t *request, void *data)
                 }
 
                 sds_metric = flb_sds_cat(sds_metric, "fluentbit_", 10);
```

Comment:
Could you replace this with a single call to `flb_sds_printf` or at least use `flb_sds_cat_safe`? Otherwise these will exchange a possible `NULL` deref for a memory leak.

Reply:
I replaced it with `flb_sds_cat_safe`